🐛 BUG: api.error.service_not_found when publishing a development worker

### What version of `Wrangler` are you using?

2.1.0

### What operating system are you using?

Linux + Github Actions

### Describe the Bug

It appears `wrangler publish` broke in the recent version. 

The error I'm getting is:
```
✘ [ERROR] A request to the Cloudflare API (/accounts/ACCOUNTID/workers/services/githubrepo-feature-refs-pull-46-merge) failed.

  workers.api.error.service_not_found [code: 10090]
  
  If you think this is a bug, please open an issue at:
  https://github.com/cloudflare/wrangler2/issues/new/choose
```

I'm getting this from just running the command `npx wrangler@2 publish --name=test123` too.

When this happens the error is being thrown from [this line](https://github.com/cloudflare/wrangler2/compare/wrangler%402.0.29...wrangler%402.1.0#diff-fa9a52007c2b6571335f75c75e00cf783306b826a082fe144ac0337b3bfff9fcR522). I'm trying to understand the difference between a script and a service (and it's API counterparts), but I can't really seem to figure out what I'm doing wrong

--- a/.changeset/cyan-cherries-stare.md
+++ b/.changeset/cyan-cherries-stare.md
@@ -1,0 +1,10 @@
+---
+"wrangler": patch
+---
+
+fix: Publish error when no upstream Worker existed
+try/catch handles the check for last deployed source.
+The check breaking when no available Worker was upstream to check,
+which was causing deployments of new workers to fail.
+
+fixes #1824

--- a/.changeset/cyan-cherries-stare.md
+++ b/.changeset/cyan-cherries-stare.md
@@ -2,8 +2,10 @@
 "wrangler": patch
 ---
 
-fix: Publish error when no upstream Worker existed
-Add a try/catch when checking when the worker was last deployed
+fix: Publish error when deploying new Workers
+
+This fix adds a try/catch when checking when the Worker was last deployed.
+
 The check was failing when a Worker had never been deployed, causing deployments of new Workers to fail.
 
 fixes #1824

--- a/.changeset/cyan-cherries-stare.md
+++ b/.changeset/cyan-cherries-stare.md
@@ -5,6 +5,5 @@
 fix: Publish error when no upstream Worker existed
 Add a try/catch when checking when the worker was last deployed
 The check was failing when a Worker had never been deployed, causing deployments of new Workers to fail.
-which was causing deployments of new workers to fail.
 
 fixes #1824

--- a/.changeset/cyan-cherries-stare.md
+++ b/.changeset/cyan-cherries-stare.md
@@ -3,8 +3,8 @@
 ---
 
 fix: Publish error when no upstream Worker existed
-try/catch handles the check for last deployed source.
-The check breaking when no available Worker was upstream to check,
+Add a try/catch when checking when the worker was last deployed
+The check was failing when a Worker had never been deployed, causing deployments of new Workers to fail.
 which was causing deployments of new workers to fail.
 
 fixes #1824

--- a/packages/wrangler/src/__tests__/publish.test.ts
+++ b/packages/wrangler/src/__tests__/publish.test.ts
@@ -6488,6 +6488,38 @@ addEventListener('fetch', event => {});`
 		`);
 		});
 	});
+
+	it("should publish if the last deployed source check fails", async () => {
+		writeWorkerSource();
+		writeWranglerToml();
+		mockSubDomainRequest();
+		mockUploadWorkerRequest();
+		setMockRawResponse(
+			"/accounts/:accountId/workers/scripts/:scriptName",
+			"PUT",
+			() => {
+				return createFetchResult({}, true, [
+					{
+						code: 10090,
+						message: "workers.api.error.service_not_found",
+					},
+				]);
+			}
+		);
+
+		await runWrangler("publish index.js");
+		expect(std).toMatchInlineSnapshot(`
+		Object {
+			"debug": "",
+			"err": "",
+			"out": "Total Upload: xx KiB / gzip: xx KiB
+		Uploaded test-name (TIMINGS)
+		Published test-name (TIMINGS)
+			https://test-name.test-sub-domain.workers.dev",
+			"warn": "",
+		}
+	`);
+	});
 });
 
 /** Write mock assets to the file system so they can be uploaded. */

--- a/packages/wrangler/src/__tests__/publish.test.ts
+++ b/packages/wrangler/src/__tests__/publish.test.ts
@@ -6510,13 +6510,13 @@ addEventListener('fetch', event => {});`
 		await runWrangler("publish index.js");
 		expect(std).toMatchInlineSnapshot(`
 		Object {
-			"debug": "",
-			"err": "",
-			"out": "Total Upload: xx KiB / gzip: xx KiB
+		  "debug": "",
+		  "err": "",
+		  "out": "Total Upload: xx KiB / gzip: xx KiB
 		Uploaded test-name (TIMINGS)
 		Published test-name (TIMINGS)
-			https://test-name.test-sub-domain.workers.dev",
-			"warn": "",
+		  https://test-name.test-sub-domain.workers.dev",
+		  "warn": "",
 		}
 	`);
 	});

--- a/packages/wrangler/src/publish.ts
+++ b/packages/wrangler/src/publish.ts
@@ -220,21 +220,26 @@ export default async function publish(props: Props): Promise<void> {
 	// TODO: warn if git/hg has uncommitted changes
 	const { config, accountId, name } = props;
 	if (accountId && name) {
-		const serviceMetaData = await fetchResult(
-			`/accounts/${accountId}/workers/services/${name}`
-		);
-		const { default_environment } = serviceMetaData as {
-			default_environment: {
-				script: { last_deployed_from: "dash" | "wrangler" | "api" };
+		try {
+			const serviceMetaData = await fetchResult(
+				`/accounts/${accountId}/workers/services/${name}`
+			);
+			const { default_environment } = serviceMetaData as {
+				default_environment: {
+					script: { last_deployed_from: "dash" | "wrangler" | "api" };
+				};
 			};
-		};
 
-		if (
-			(await fromDashMessagePrompt(
-				default_environment.script.last_deployed_from
-			)) === false
-		)
-			return;
+			if (
+				(await fromDashMessagePrompt(
+					default_environment.script.last_deployed_from
+				)) === false
+			)
+				return;
+		} catch (e) {
+			if ((e as { code?: number }).code === 10090) return;
+			logger.error(e);
+		}
 	}
 
 	if (!(props.compatibilityDate || config.compatibility_date)) {

